addons/mail/static/src/components/chat_window_manager/chat_window_manager_tests.js

```diff
@@ -671,6 +671,66 @@ QUnit.test('chat window: close on ESCAPE', async function (assert) {
     assert.verifySteps(['rpc:channel_fold/closed']);
 });
 
+QUnit.test('focus next visible chat window when closing current chat window with ESCAPE', async function (assert) {
+    assert.expect(4);
+
+    Object.assign(this.data.initMessaging, {
+        channel_slots: {
+            channel_channel: [
+                {
+                    channel_type: 'channel',
+                    id: 20,
+                    is_minimized: true,
+                    is_pinned: true,
+                    name: "General",
+                    state: 'open',
+                    uuid: 'channel-20-uuid',
+                },
+                {
+                    channel_type: 'channel',
+                    id: 21,
+                    is_minimized: true,
+                    is_pinned: true,
+                    name: "other chat",
+                    state: 'open',
+                    uuid: 'channel-21-uuid',
+                }
+            ],
+        },
+    });
+
+    await this.start();
+
+    // Assert than the 2 windows exists
+    assert.strictEqual(
+        document.querySelectorAll(".o_ChatWindow").length,
+        2,
+        "2 chat windows should be present"
+    );
+    // No window should be folded
+    assert.containsNone(
+        document.body,
+        '.o_ChatWindow .o_folded',
+        "All chat window should be open"
+    );
```

Comment:
Note: one last init assert that would be nice to have is one to ensure no chat window is initially `o-focused`, to ensure there is indeed a change of focus that happens after Escape.

Reply:
But the first window should be focused, otherwise, pressing the escape key will not trigger the close.
So we should test that one chat window is focused

Reply:
I don't think the focus is necessary before in tests due to the way we trigger the event manually on a specific element. But indeed in a real situation it would have the focus.

My suggestion was to have the simplest test, but if you really want to go all the way, you need to ensure the the focus is initially on a specific chat window, that it is the one being closed, and then that the focus is on the other chat window specifically.

Reply:
Since we dispatch the event directly on the `.o_ComposerTextInput_textarea`, there is no chat window focused from the init.
Look hard to test properly (and maybe not necessary)